src/leveled_setoplexer.xrl

```diff
@@ -0,0 +1,24 @@
+Definitions.
+WhiteSpace  = ([\t\f\v\r\n\s]+)
+
+Rules.
+
+{WhiteSpace} : skip_token.
+
+\(        : {token, {'(', TokenLine}}.
+\)        : {token, {')', TokenLine}}.
+
+OR        : {token, {'UNION', TokenLine}}.
+AND       : {token, {'INTERSECT', TokenLine}}.
+NOT       : {token, {'SUBTRACT', TokenLine}}.
```

Comment:
Kind, of, NOT is in fact a unary operation and SUBTRACT a binary.

This might cause future confusion. If you write NOT $d, then you mean to say the universe apart from whatever is in $d. But the universe is not part of your language.

- But more prominently, if I write `$a SUBTRACT NOT $b` you fail, whereas in set theory you would take the complenet from B and subtract that from A.

It will be hard to implement complement. So not sure you want to add NOT. Also because NOT is very hard to do in regular expressions.